docs/source/en/tasks/image_classification.mdx

```diff
@@ -228,6 +269,144 @@ Once training is completed, share your model to the Hub with the [`~transformers
+```
+
+To compute the accuracy from the predictions use a [Keras callback](https://huggingface.co/docs/transformers/main/en/tasks/./main_classes/keras_callbacks).
+Pass your `compute_metrics` function to [KerasMetricCallback](https://huggingface.co/docs/transformers/main/en/main_classes/keras_callbacks#transformers.KerasMetricCallback):
+
+```py
+>>> from transformers.keras_callbacks import KerasMetricCallback
+
+>>> metric_callback = KerasMetricCallback(metric_fn=compute_metrics, eval_dataset=tf_eval_dataset)
+>>> callbacks = [metric_callback]
+```
+
+Finally, you are ready to train your model! Call `fit()` with your training and validation datasets, the number of epochs,
+and your callbacks to fine-tune the model:
+
+```py
+>>> model.fit(tf_train_dataset, validation_data=tf_eval_dataset, epochs=num_epochs, callbacks=callbacks)
+Epoch 1/5
+250/250 [==============================] - 313s 1s/step - loss: 2.5623 - val_loss: 1.4161 - accuracy: 0.9290
+Epoch 2/5
+250/250 [==============================] - 265s 1s/step - loss: 0.9181 - val_loss: 0.6808 - accuracy: 0.9690
+Epoch 3/5
+250/250 [==============================] - 252s 1s/step - loss: 0.3910 - val_loss: 0.4303 - accuracy: 0.9820
+Epoch 4/5
+250/250 [==============================] - 251s 1s/step - loss: 0.2028 - val_loss: 0.3191 - accuracy: 0.9900
+Epoch 5/5
+250/250 [==============================] - 238s 949ms/step - loss: 0.1232 - val_loss: 0.3259 - accuracy: 0.9890
+```
+
+In order to push your trained Keras model to Hub, you need to build it first. When building a Keras model, you must
+provide the model's input shape. In this case, we can get it using image processor's attributes. Note that depending on
+the pretrained model you are fine-tuning, the corresponding image processors may have other dictionary keys
+(e.g. `shortest_edge` instead of `height` and `width`).
+
+```py
+>>> from huggingface_hub import push_to_hub_keras
+
+>>> model.build((None, None, image_processor.size["height"], image_processor.size["width"]))
+>>> output_dir = "food_classifier"
+>>> push_to_hub_keras(model, repo_id=f"MariaK/{output_dir}")
+```
+
+Finally, if you want to use this model for inference with [`pipeline`], or re-use it for further fine-tuning,
+you need to upload the following files to Hub:
+- `preprocessor_config.json` that you can save from the image_processor you have used during the training
+- model's weights in `h5` format, and model's `config.json` that you can save by calling the `save_pretrained()` method on the model
+
+```py
+>>> from huggingface_hub import HfApi
+>>> from huggingface_hub import HfFolder
+
+>>> api = HfApi()
+>>> hub_token = HfFolder.get_token()
+
+>>> image_processor.save_pretrained(output_dir)
+>>> model.save_pretrained(output_dir)
+
+>>> api.upload_folder(
+...     token=hub_token,
+...     folder_path=output_dir,
+...     path_in_repo="",
+...     repo_id=f"MariaK/{output_dir}",
+...     ignore_patterns="**/logs/*.txt",
+... )
+```
```

Comment:
This is way too complicated compared to using the `PushToHubCallback`. Why not use it? The only weird thing is that the `image_processor` should be passed to the `tokenizer` argument (which might need to be renamed to `processor` at some point).

Reply:
This is the reason why I am not using `PushToHubCallback` at the moment, it hangs the cell execution. https://github.com/huggingface/transformers/issues/21116

Reply:
Let's remember to move back to it when the bug is fixed then :-)

Reply:
So according to the feedback on the issue, `PushToHubCallback` partly works. If save_strategy is set to "no", it will push the model on the train completion, so I switched to the callback. 
Once the issue is resolved I can remove the save_strategy parameter. Please review. 63f1d9fdfc546c40f1e3cfbcb8e29f9a5787ae1f